common/go-utils/src/vrs.rs

```diff
@@ -57,75 +34,57 @@ impl VersionRequirement<GoVersion> for GoRequirement {
     fn parse(input: &str) -> Result<Self, RequirementParseError> {
         input
             .strip_prefix("go")
-            .map_or(parse_semver_requirement(input), |req| {
-                parse_semver_requirement(format!("={req}").as_str())
+            .map_or(SemanticVersionRequirement::parse(input), |req| {
+                SemanticVersionRequirement::parse(format!("={req}").as_str())
             })
-            .map(Self)
+            .map(GoRequirement)
     }
 }
 
-fn parse_semver_requirement(input: &str) -> Result<semver::VersionReq, RequirementParseError> {
-    semver::VersionReq::parse(input).map_err(RequirementParseError)
-}
-
 impl fmt::Display for GoRequirement {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         write!(f, "{}", self.0)
     }
 }
 
-/// `Version` is a wrapper around `semver::Version` that adds
-/// - `Deserialize` and `Serialize` traits
-/// - Ability to parse go-flavored versions
+/// `GoVersion` is a wrapper around `SemanticVersion` that adds
+///  ability to parse go-flavored versions
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, PartialOrd, Ord)]
-#[serde(try_from = "String", into = "String")]
-pub struct GoVersion(semver::Version);
+pub struct GoVersion(SemanticVersion);
 
 #[derive(thiserror::Error, Debug)]
-pub enum VersionParseError {
+pub enum GoVersionParseError {
     #[error("Couldn't parse go version: {0}")]
     SemVer(#[from] semver::Error),
     #[error("Internal buildpack issue parsing go version regex: {0}")]
     Regex(#[from] regex::Error),
     #[error("Couldn't parse version. Unable to capture values from regex.")]
     Captures,
+    #[error("Couldn't parse go version: {0}")]
+    SemanticVersion(#[from] SemanticVersionParseError),
```

Comment:
Shouldn't this replace the `SemVer` error above? Are both required?

Reply:
You're right -- this has been resolved by removing the `SemanticVersion` abstraction altogether :)